backend/routes/auth.py

```
@@ -136,39 +167,58 @@ def get_github_user(access_token: str) -> dict:
+        "iat": now,
+    }
+    return jwt.encode(payload, secret, algorithm=JWT_ALGORITHM)
+
+
+def verify_magic_link_token(token: str) -> str:
+    """Verify a magic link token and return the authorized admin email."""
+    secret = _get_session_secret()
+    try:
+        payload = jwt.decode(token, secret, algorithms=[JWT_ALGORITHM])
+    except jwt.ExpiredSignatureError:
+        raise HTTPException(status_code=400, detail="Magic link expired")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)